mkosi/__init__.py

```diff
@@ -352,9 +362,12 @@ def finalize_scripts(scripts: Mapping[str, Sequence[PathString]] = {}) -> Iterat
         yield Path(d)
 
 
-def finalize_host_scripts(state: MkosiState, chroot: Sequence[PathString]) -> contextlib.AbstractContextManager[Path]:
+def finalize_host_scripts(
+        state: MkosiState,
+        helpers: Mapping[str, Sequence[PathString]],
+) -> contextlib.AbstractContextManager[Path]:
     git = {"git": ("git", "-c", "safe.directory=*")} if find_binary("git") else {}
-    return finalize_scripts(git | {"mkosi-chroot": chroot} | package_manager_scripts(state))
+    return finalize_scripts(git | helpers | package_manager_scripts(state))
```

Comment:
pyright complains about the type here. There are two workaround:
1. typing `helpers` in the signature as `dict` instead of `Mapping`
2. adding a cast around helpers on this line

Though we strive for generic things here and `Mapping` does imply this being a readonly in addition, I think practically beats purity here, though a TODO comment for the future to look at this again would make sense.